components/layout_2020/flexbox/layout.rs

```diff
@@ -1957,20 +1957,22 @@ impl FlexItem<'_> {
         let item_style = independent_formatting_context.style();
         match &independent_formatting_context.contents {
             IndependentFormattingContextContents::Replaced(replaced) => {
+                let min_size = flex_axis.vec2_to_flow_relative(self.content_min_size);
+                let max_size = flex_axis.vec2_to_flow_relative(self.content_max_size);
                 let size = replaced.used_size_as_if_inline_element_from_content_box_sizes(
                     containing_block,
                     item_style,
                     self.preferred_aspect_ratio,
-                    LogicalVec2 {
-                        inline: Size::Numeric(inline_size),
-                        block: block_size.non_auto().map_or(Size::Initial, Size::Numeric),
-                    },
-                    flex_axis
-                        .vec2_to_flow_relative(self.content_min_size)
-                        .map(|size| Size::Numeric(*size)),
-                    flex_axis
-                        .vec2_to_flow_relative(self.content_max_size)
-                        .map(|size| size.map_or(Size::Initial, Size::Numeric)),
+                    &Sizes::new(
+                        block_size.non_auto().map_or(Size::Initial, Size::Numeric),
+                        Size::Numeric(min_size.block),
+                        max_size.block.map_or(Size::Initial, Size::Numeric),
+                    ),
+                    &Sizes::new(
+                        Size::Numeric(inline_size),
+                        Size::Numeric(min_size.inline),
+                        max_size.inline.map_or(Size::Initial, Size::Numeric),
+                    ),
```

Comment:
@Loirooriol I'm referring to things like this code. Does the replaced box not know it's own `min-` and `max-` sizes?

I realise this isn't introduced in this PR. But I was just wondering whether we could eliminate these parameters entirely so that we don't need a new type to cut down on the boilerplate.

Reply:
Well, yes, that's what `used_size_as_if_inline_element()` does: it takes the computed values from the style and resolves them with the containing block.

In this case flexbox is using `used_size_as_if_inline_element_from_content_box_sizes` which is like a lower-level API. I think the reasons were that it already has the data (so no need to resolve percentages again and such) and also to handle automatic minimums, which are typically zero but not in flexbox.

That said I still haven't implemented sizing keywords for flexbox so this may need to change a bit.

Reply:
> I think the reasons were that it already has the data (so no need to resolve percentages again and such) and also to handle automatic minimums, which are typically zero but not in flexbox.

I think I'd be more inclined to keep the interface simple here as I feel like the performance gains are probably quite minor. But I we sort this out (either way) later :)